src/Service/Connector/NamecheapConnector.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace App\Service\Connector;
+
+use App\Entity\Domain;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autoconfigure;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+#[Autoconfigure(public: true)]
+class NamecheapConnector extends AbstractProvider
```

Comment:
Perhaps change the class name for `NamecheapProvider` to be consistent with the other classes.

```suggestion
class NamecheapProvider extends AbstractProvider
```